src/ert/config/ert_config.py

```diff
@@ -883,6 +884,12 @@ def from_dict(cls, config_dict) -> Self:
                 if not isinstance(config, GenKwConfig):
                     continue
                 group_params = {x.name for x in config.transform_function_definitions}
+                if group_name == DESIGN_MATRIX_GROUP:
+                    dm_errors.append(
+                        ConfigValidationError(
+                            f"Cannot have GEN_KW with group name {DESIGN_MATRIX_GROUP} when using DESIGN_MATRIX keyword."
```

Comment:
I don't like the wording tbh.
But couldn't come up with a better one. So good :)